libs/api/utils/src/lib/api-utils.ts

```diff
@@ -1,21 +1,53 @@
 import { Prisma, prisma } from '@nmit-coursition/db'
-import { isDateBeforeNow } from '@nmit-coursition/utils'
-import { parseApiKey } from '../api'
+import { generateRandomIdentifier, isDateBeforeNow } from '@nmit-coursition/utils'
+import { Elysia } from 'elysia'
+import { formatApiErrorResponse, parseApiKey } from '../api'
 import type { ApiErrorCode } from '../errorList'
-import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest } from '../typescript'
+import { ERROR_LIST } from '../errorList'
+import { errorResponseModel } from '../model'
+import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest, ExtendedRequest } from '../typescript'
 
 let API_KEY_TO_ID_CACHE: { [key: string]: bigint } = {}
 
-let API_KEY_CONTEXT: string | undefined
+export const apiCommonGuard = new Elysia().guard({
+  as: 'global',
```

Comment:
**thought:** If we change from `global` to `scoped` we might not need `if (path.startsWith('/swagger')) return`.